inception-project/inception
# #1239 - Support recommendation of simple relations

# What's in the PR
- Rename String Recommender package
- Start simple projecting relation recommender
- Renamed `AnnotationSuggestion` to `SpanSuggestion`
- Introduced `AnnotationSuggestion` interface on top of `AnnotationSuggestion_ImplBase`

# How to test manually  

_**This PR removes columns from the database, switching to another branch after running this might be difficult before it is merged!**_

* Create a new project and upload a document
* Define a new span layer with a String feature 
* Add a relation layer ontop of that previously created layer and add a relation layer
* Add a string relation recommender on that relation layer
* Mark a span "a" and a span "b" in the same sentence
* Draw a relation between them and give it a label
* In another sentence, also mark a span "a" and a span "b" in the same sentence
* It should recommend the label you defined for the first sentence

* Check that active learning and recommender still work
* Check that existing string recommender in other projects still work

**Automatic testing**

**Documentation**

**Things missing:**

- [x]  Check active learning
- [x] Sorting by confidence
- [x] Max suggestions
- [x] Rejection hiding and learning record
- [x] Lazy details (confidence, label)
- [x] Several recommender grouping
